src/modules/UI/components/Modals/InteractiveModal/InteractiveModal.ui.js

```diff
@@ -0,0 +1,205 @@
+// @flow
+
+import React, { Component } from 'react'
+import type { Node } from 'react'
+import { StyleSheet, Text, View } from 'react-native'
+
+import { default as Modal } from 'react-native-modal'
+import { styles } from './styles.js'
+
+// CONTAINER /////////////////////////////////////////////////////////////////////////////
+export type ContainerProps = {
+  children: Node,
+  style?: StyleSheet.Styles
+}
+export class Container extends Component<ContainerProps> {
+  render () {
+    return (
+      <View {...this.props} style={[styles.container, this.props.style]}>
+        {this.props.children}
+      </View>
+    )
+  }
+}
+
+// HEADER /////////////////////////////////////////////////////////////////////////////
+export type HeaderProps = {
+  children: Node,
+  style?: StyleSheet.Styles
+}
+export class Header extends Component<HeaderProps> {
+  render () {
+    return (
+      <View {...this.props} style={[styles.header, this.props.style]}>
+        {this.props.children}
+      </View>
+    )
+  }
+}
+
+// ANDROID_HACK_SPACER /////////////////////////////////////////////////////////////////////////////
+/*
+  This spacer should be used with Icon to overcome the limitations on Android
+  React Native on Android does not support 'Overflow'
+  If/When React Native on Android supports 'Overflow',
+    * remove the hack component
+    * move the Icon component inside Modal.Container
+  https://github.com/facebook/react-native/issues/6802
+*/
+type AndroidHackSpacerProps = {
+  style?: StyleSheet.Styles
+}
+export class AndroidHackSpacer extends Component<AndroidHackSpacerProps> {
+  render () {
+    return (
+      <View style={styles.androidHackSpacer} />
+    )
+  }
+}
+
+// ICON /////////////////////////////////////////////////////////////////////////////
+export type IconProps = {
+  children: Node,
+  style?: StyleSheet.Styles
+}
+export class Icon extends Component<IconProps> {
+  static AndroidHackSpacer = AndroidHackSpacer
+  render () {
+    return (
+      <View {...this.props} style={[styles.icon, this.props.style]}>
+        {this.props.children}
+      </View>
+    )
+  }
+}
+
+// TITLE /////////////////////////////////////////////////////////////////////////////
+type TitleProps = {
+  children: Node,
+  style?: StyleSheet.Styles
+}
+export class Title extends Component<TitleProps> {
+  render () {
+    return (
+      <Text style={[styles.title, this.props.style]} {...this.props}>
```

Comment:
React Native cascades text styles, so this would be valid, and styled...
```
<Text style={styles.title}>
  <Text>{TITLE_TEXT}</Text>
</Text>
```
If something other than `<Text>` were used, i think it might display a warning, not sure though